service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java

```diff
@@ -230,17 +237,44 @@ void shouldReturnMostRecentUploadedOrderDocumentUrl() {
             returnedMostRecentUploadedOrderDocumentUrl);
     }
 
-    private void assertCommonC21Fields(GeneratedOrder order) {
-        assertThat(order.getType()).isEqualTo(BLANK_ORDER);
-        assertThat(order.getDocument()).isEqualTo(DocumentReference.builder().build());
-        assertThat(order.getDetails()).isEqualTo("Some details");
-        assertThat(order.getDate()).isNotNull();
-        assertThat(order.getJudgeAndLegalAdvisor()).isEqualTo(JudgeAndLegalAdvisor.builder().build());
+    @Test
+    void shouldRemoveOrderPropertiesWhenTheyExistInCaseDetails() {
+        Map<String, Object> data = new HashMap<>();
+        data.put("orderTypeAndDocument", "some data");
+        data.put("order", "some data");
+        data.put("judgeAndLegalAdvisor", "some data");
+        data.put("orderFurtherDirections", "some data");
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(data)
+            .build();
+
+        service.removeOrderProperties(caseDetails);
+        assertThat(caseDetails.getData()).isEmpty();
+    }
+
+    private static Stream<Arguments> fileNameSource() {
+        return Stream.of(
+            Arguments.of(OrderTypeAndDocument.builder().type(BLANK_ORDER).build(), "blank_order_c21.pdf"),
```

Comment:
Type `CARE_ORDER` with `null` subtype should also be checked.

Reply:
added a check